core/src/test/java/com/nuodb/migrator/globalStore/GlobalStoreTest.java

```diff
@@ -0,0 +1,182 @@
+import static com.nuodb.migrator.jdbc.session.SessionUtils.createSession;
+import static org.mockito.BDDMockito.*;
+import static org.mockito.Mockito.when;
+import static com.nuodb.migrator.jdbc.metadata.DatabaseInfos.NUODB;
+import static org.testng.Assert.*;
+
+import java.sql.DatabaseMetaData;
+import java.util.Collection;
+import java.util.Map;
+
+import static org.mockito.Mockito.mock;
+
+import com.nuodb.migrator.spec.DumpJobSpec;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import org.mockito.Spy;
+import com.nuodb.migrator.job.JobExecution;
+import com.nuodb.migrator.job.JobExecutor;
+
+import static com.google.common.collect.Maps.newHashMap;
+import static com.nuodb.migrator.job.JobExecutors.createJobExecutor;
+
+import com.nuodb.migrator.jdbc.dialect.NuoDBDialect;
+
+//@SuppressWarnings("all")
+public class GlobalStoreTest {
+
+    private GlobalStore globalStore;
+    private Table table;
+    @Mock
+    public LoadTable mockedLoadTable;
+    private ScriptGeneratorManager scriptGeneratorManager;
+    private BackupLoader backupLoader;
+    @Spy
+    @InjectMocks
+    private DumpJob dumpJob = new DumpJob(new DumpJobSpec());
+    private JobExecutor jobExecutor;
+    private Map<Object, Object> jobContext;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        initMocks(this);
+        GlobalStore globalStore = GlobalStore.getInstance();
+        DatabaseMetaData databaseMetaData = mock(DatabaseMetaData.class);
+        given(databaseMetaData.getDatabaseProductName()).willReturn("NuoDB");
+
+    }
+
+    @DataProvider(name = "alterScript")
+    public Object[][] testAlterScriptWithGeneratedAlwaysTrue() {
+        GlobalStore gb = GlobalStore.getInstance();
+
+        Dialect dialect = new NuoDBDialect();
+        Schema schema = new Schema("schema1");
+        Table table = new Table("test_table");
+        Column column = table.addColumn("id");
+        column.setTypeCode(BIGINT);
+        column.setTypeName("BIGINT");
+        column.setNullable(false);
+        column.setPosition(1);
+        Sequence sequence = createSequence(null, "schema1", "test_table", "id");
+        sequence.setName("sequence");
+        column.setSequence(sequence);
+        schema.addSequence(sequence);
+        column.getSequence().setGeneratedAlways(true);
+        table.addColumn(column);
+        schema.addTable(table);
+        gb.put(sequence.getName(), sequence.isGeneratedAlways());
+        String alterStat = "ALTER TABLE " + schema.getName() + "." + table.getName() + " MODIFY " + column.getName()
+                + " " + column.getJdbcType().getTypeName() + " GENERATED ALWAYS AS IDENTITY";
+
+        return new Object[][] { { table, gb, dialect, column, alterStat } };
+    }
+
+    @Test(dataProvider = "alterScript")
+    public void testGeneratedAlwaysTrue(Table table, GlobalStore gb, Dialect dialect, Column col, String expected) {
```

Comment:
This method, `testGeneratedAlwaysFalse`, and `testWithoutGeneratedAlways` are all identical and they all reference the same specified on the method `testAlterScriptWithGeneratedAlwaysTrue`.

They do not actually test what the method name says they test, they only exercise the one set of test values provided by `testAlterScriptWithGeneratedAlwaysTrue`, three different times. Basically, this is running the same test three times, with two of them being named incorrectly.

Please just remove all of the indirection having to do with `@DataProvider`, which is only adding confusion and making the test not actually do what you what. Besides the `setUp()` method, there should be only three methods in this class:

```java
    @Test
    public void testAlterScriptWithGeneratedAlwaysTrue() {
        ...
        LoadTable tb = new LoadTable(null, table, null);
        String actualScripts = gb.alterScript(tb);
        assertEquals(expected, actualScripts);
    }

    @Test
    public void testAlterScriptWithGeneratedAlwaysFalse() {
        ...
        LoadTable tb = new LoadTable(null, table, null);
        String actualScripts = gb.alterScript(tb);
        assertEquals(expected, actualScripts);
    }

    @Test
    public void testAlterScriptWithoutGeneratedAlways() {
        ...
        LoadTable tb = new LoadTable(null, table, null);
        String actualScripts = gb.alterScript(tb);
        assertEquals(expected, actualScripts);
    }
```